otherlibs/win32unix/unix.ml

```diff
@@ -457,6 +458,21 @@ external chdir : string -> unit = "unix_chdir"
 external getcwd : unit -> string = "unix_getcwd"
 let chroot _ = invalid_arg "Unix.chroot not implemented"
 
+let unlink path =
+  (* On Windows, trying to unlink a symlink to a directory will
+     raise, but the symlink can be deleted with rmdir. *)
+  try unlink path with
+  | Unix_error (EACCES, _, _) as exn_unlink ->
+    let {st_kind; _} = try lstat path with _ -> raise exn_unlink in
+    begin match st_kind with
+    | S_DIR ->
+      (* POSIX uses EPERM rather than EACCES when the path is a
+       * directory. *)
+      raise (Unix_error (EPERM, path, error_message EPERM))
+    | S_LNK -> rmdir path
+    | _ -> raise exn_unlink
+    end
```

Comment:
```suggestion
    if st_kind = S_LNK then
      try rmdir path with _ -> raise e
    else
      raise e
```
(the key point is that `rmdir` will raise if the user actually doesn't have permission to delete the symlink, so you want to avoid leaking the `Unix.Unix_error(Unix.EACCES, "rmdir", "source")`)

Reply:
I thought that in that case it was more interesting to have the error that the user doesn't have the right permissions instead of the unlink error just telling us we're not using the right function to delete the entry.
I've seen that generally we want to re-raise the first error encountered, but here isn't the rmdir more relevant?